src/uthash.h

```diff
@@ -177,24 +193,45 @@ do {
 
 #define HASH_MAKE_TABLE(hh,head)                                                 \
 do {                                                                             \
+  int _mem_ok = 1;                                                               \
   (head)->hh.tbl = (UT_hash_table*)uthash_malloc(sizeof(UT_hash_table));         \
-  if (!(head)->hh.tbl) {                                                         \
+  if (!((head)->hh.tbl)) {                                                       \
+    if (HASH_NOMEM_OK) {                                                         \
+      break;                                                                     \
+    }                                                                            \
     uthash_fatal("out of memory");                                               \
```

Comment:
I'm skeptical of the way this patch *avoids* calling `uthash_fatal` in OOM situations. Shouldn't we notify the client *somehow* that we've hit OOM, and then leave it up to the client ('s `uthash_fatal` handler) to say "exit" or "longjmp" or "just keep going"?